src/generate/utils.ts

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { model } from "amf-client-js";
+
+export const ARRAY_DATA_TYPE = "Array";
+export const DEFAULT_DATA_TYPE = "any";
+export const OBJECT_DATA_TYPE = "object";
+export const PRIMITIVE_DATA_TYPE_MAP = {
+  "http://www.w3.org/2001/XMLSchema#string": "string",
+  "http://www.w3.org/2001/XMLSchema#integer": "number",
+  "http://www.w3.org/2001/XMLSchema#double": "number",
+  "http://www.w3.org/2001/XMLSchema#float": "number",
+  "http://www.w3.org/2001/XMLSchema#boolean": "boolean"
+};
+
+export const getPayloadResponses = (
+  operation: model.domain.Operation
+): model.domain.Response[] => {
+  const okResponses = [];
+  for (const res of operation.responses) {
+    if (res.statusCode.nonEmpty && res.statusCode.value().startsWith("2")) {
+      okResponses.push(res);
+    }
+  }
+  return okResponses;
+};
+
+/**
+ * Given a payload, extract the types.
+ *
+ * @param payload - Contains schema(s) from which to extract the type(s).
+ * @returns string representation of the datatypes in the payload
+ */
+export const extractTypeFromPayload = (
```

Comment:
All other methods seem to have the nomenclature of `getFoosBar`  I kind of like `getBarFromFoo`  but this seems to be the only one saying 'extract'